### DISCUSSION TITLE
Loop issue with Fast Refresh on the app generated with prebuild

### DISCUSSION THREAD

**Discussion by vgo-hub** [expo/expo#29039] [Archived: Questions]
Hello everyone,

I'm encountering an issue with Fast Refresh on the app generated with the prebuild of my project. Despite not making any changes to the code, the process enters an infinite loop. Unfortunately, I can't share the code for detailed review.

However, I tried starting from the initial basic _layout, removing all the code and keeping it simple as if it were a new project, but the loop still occurs.

Does anyone have any idea what might be causing this issue or suggestions on how to fix it?

Thanks in advance for the help!

https://github.com/expo/expo/assets/131958460/65f112f0-42ba-4e44-ad85-fb5e6d9159c0

**Comment by eyedroot** [comment]
I'm having the same problem.

**Comment by ahmtxonly** [comment]
same

**Comment by raajnadar** [comment]
It seems a fix is added to the main branch - https://github.com/expo/expo/pull/29157

**Comment by eyedroot** [comment]
@veigoo-repo Did this solve your problem? I'm still having the same issue.

**Reply by vgo-hub** [reply]
Not yet, because every time the issue occurs, I terminate the Metro Bundler and restart it. Also, it's so random that I haven't been able to identify it. However, it should be related to @raajnadar point; I need to update to the latest version, which is 51.0.9. Try it and let me know.

**Reply by eyedroot** [reply]
I'm using 51.0.9 and it happens every time, so I've turned off Fast Reload to develop, it's too stressful. I'm even considering downgrading my expo-router.

**Reply by vgo-hub** [reply]
I imagine. It happens much less to me now, but I haven't figured out why, mainly when I modify files inside the app folder. Try using watchman watch-del... and watchman watch-project... The issue will surely reappear, but it might lessen for a while. It's unclear where the problem is.

**Reply by eyedroot** [reply]
thanks!

**Reply by eyedroot** [reply]
I'm still having the same problem with the `watchman watch-del.. and watchman watch-project` method.

I've even tried completely deleting `watchman` and reinstalling it, but the same problem persists.

Have you tried anything else besides watchman!?

**Reply by vgo-hub** [reply]
No, I only tried this and also deleting Watchman like you did. When it occurs, I stop Metro and restart it. It seems to work until it happens again, then I run the same command. Does it happen frequently to you? We should open an issue, but I don't have a demonstration. If I have time and can manage, or maybe if you can, try to check if the modification at @raajnadar  point that was mentioned is actually present in the installed Expo package. If not, we need to try applying the changes manually and see if it resolves the issue.

**Reply by eyedroot** [reply]
Oops, as you said, I opened the `node_modules` file and checked the build file and there was no `shouldRunAgain = false;` code.

Do you happen to know why there is no code, as I am currently using the latest version of expo-router, 3.5.14?

I'm currently manually putting the code in and restarting it and it's working fine.

![image](https://github.com/expo/expo/assets/17328737/2ff9c28a-c7f1-474c-85c7-ad1ecb057993)

Below is the expo-router part of my yarn.lock file.

```
expo-router@~3.5.11:
  version "3.5.14"
  resolved "https://registry.yarnpkg.com/expo-router/-/expo-router-3.5.14.tgz#5ca0e861271fa716b94a6149d696f7ac1258aad4"
  integrity sha512-RVsyJLosZSq89ebA5qfPToRcFZJoTTb+6nwPHk6iESD6KON/Q7ZODt5fvCwXkY86tLNEMGo160QvBPfBZmglaA==
  dependencies:
    "@expo/metro-runtime" "3.2.1"
    "@expo/server" "^0.4.0"
    "@radix-ui/react-slot" "1.0.1"
    "@react-navigation/bottom-tabs" "~6.5.7"
    "@react-navigation/native" "~6.1.6"
    "@react-navigation/native-stack" "~6.9.12"
    expo-splash-screen "0.27.4"
    react-native-helmet-async "2.0.4"
    schema-utils "^4.0.1"
```

**Reply by vgo-hub** [reply]
Fantastic! It probably hasn't been merged with the main branch yet, but I'm not sure, I haven't checked thoroughly. I'll apply the change as well!

**Reply by vgo-hub** [reply]
No, I checked and the merge has been done. In my case, I haven't updated to the resolved version yet. In your case, I'm not sure; the version was released 5 days ago.

**Reply by eyedroot** [reply]
Yes, I noticed the code is merged too. 

If you end up updating expo-router in the future, could you please check if there is actually that code?

Finally, thank you so much for your help in solving this!

**Comment by oleg3505** [comment]
Same for me, new expo project, ios app, expo-router, and fast refresh looping.
Update: The new 3.5.15 version of expo-router fixes this issue

**Comment by felipe-software** [comment]
Still experiencing this on expo sdk 55. I think it's related to watchman watching too many files? or the Home directory. Not sure yet...
